RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,278 @@
+
+      for (uint32_t j = 0; j < DetToFeds_size; ++j) {
+        (*DetToFedsCollection)->qualityOk_(j) = detToFeds_qualityFlags[j];
+      }
+      return DetToFedsCollection;
+    }
+
+    std::unique_ptr<SiStripClusterizerConditionsDataHost> produceData(
+        SiStripClusterizerConditionsDataRecord const& iRecord) {
+      auto gains = iRecord.getTransientHandle(gainsToken_);
+      const auto& noises = iRecord.get(noisesToken_);
+      const auto& quality = iRecord.get(qualityTokenB_);
+
+      // Prepare the conditions on the host
+      // Note: most likely depending on the size of this operation, perfomance could be much likely improved moving this operation in a device kernel.
+      //       Some thinking should go into understanding how to optimize the move of gain/noise/quality and how to define a alpaka-kernel doing the
+      //       preparation for the conditions.
+      //       (same algo from RecoLocalTracker/SiStripClusterizer/plugins/SiStripClusterizerConditionsESProducer.cc)
+
+      std::vector<float> invthick(sistrip::NUMBER_OF_FEDS * sistrip::FEDCH_PER_FED);
+      std::vector<uint32_t> detID(sistrip::NUMBER_OF_FEDS * sistrip::FEDCH_PER_FED);
+      std::vector<uint16_t> iPair(sistrip::NUMBER_OF_FEDS * sistrip::FEDCH_PER_FED);
+      std::vector<uint16_t> noise(sistrip::NUMBER_OF_FEDS * sistrip::FEDCH_PER_FED * sistrip::STRIPS_PER_FEDCH);
+      std::vector<float> gain(sistrip::NUMBER_OF_FEDS * sistrip::APVS_PER_FEDCH * sistrip::FEDCH_PER_FED);
+
+      fillSiStripClusterizerConditions_(quality, gains.product(), noises, invthick, detID, iPair, noise, gain);
+
+      // LogDebug("StripCondsESProd") << "Produced a SiStripClusterizerConditions object for " << detToFeds.size()
+      //                              << " modules";
+
+      // Prepare the product & Fill the product
+      const int Data_fedch_size = detID.size();
+      const int Data_strip_size = noise.size();
+      const int Data_apv_size = gain.size();
+      //// Typical sizes for these collections (from MC run)
+      //// constexpr const unsigned int Data_fedch_size= 42240;
+      //// constexpr const unsigned int Data_strip_size= 10813440;
+      //// constexpr const unsigned int Data_apv_size= 84480;
+
+      assert(detID.size() == iPair.size() && iPair.size() == invthick.size());
+
+      // PortableHostMultiCollection(const std::array<int32_t, members_>& sizes, alpaka_common::DevHost const& host)
+      auto product = std::make_unique<SiStripClusterizerConditionsDataHost>(
+          std::array<int32_t, 3>{{Data_fedch_size, Data_strip_size, Data_apv_size}}, cms::alpakatools::host());
+
+      auto Data_fedch_View = product->view<SiStripClusterizerConditionsData_fedchSoA>();
+      auto Data_strip_View = product->view<SiStripClusterizerConditionsData_stripSoA>();
+      auto Data_apv_View = product->view<SiStripClusterizerConditionsData_apvSoA>();
+
+      for (int j = 0; j < Data_fedch_size; ++j) {
```

Comment:
I think don't understand the program logic 🤔 

From what I understand, first the values are filled in an `std::vector`, and then are copied as-is into the SoA.

If that is the case, why not fill the SoA directly ?